packages/nixos/azure.nix

```diff
@@ -85,5 +82,21 @@ in
         ExecStart = "${lib.getExe pkgs.azure-no-agent}";
       };
     };
+
+    systemd.services.setup-nat-for-imds = {
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+      description = "Setup NAT for IMDS";
+      serviceConfig = {
+        Type = "oneshot";
+        RemainAfterExit = "yes";
+        # TODO: Find out why just ordering this after network-online.target
+        # isn't sufficient. (Errors with saying that the network is unreachable)
```

Comment:
Do you know what step produces these errors? One dependency that's not declared is the `podns` existence.

Reply:
If the `netns` exists but routing is not set up yet, that would explain it.

Reply:
Verifying that hypothesis right now.

Reply:
Still fails at the very same line. Ordering is correct though:

```
...
 195ms setup-nat-for-imds.service
 193ms systemd-udevd.service
 190ms systemd-oomd.service
 154ms netns@podns.service
...
```

Reply:
setup-nat should go after netns, is that actually what your snippet shows?

Reply:
Actually, having the podns is not enough, it also needs routes set up, which might only happen in the agent-protocol-forwarder.

Reply:
Good point, it doesn't! `systemd-analyze blame` does not order by execution but by time the unit took to execute. 

However, the ordering still seems to be correct. `netns@podns.service` executes (exits) 10s before `setup-nat-for-imds.service` starts.

Reply:
Can you elaborate? This works as-is. I don't know what I should change.